mlflow/gateway/providers/bedrock.py

```diff
@@ -0,0 +1,279 @@
+            ) from e
+
+    @property
+    def _underlying_provider(self):
+        if (not self.config.model.name) or "." not in self.config.model.name:
+            return None
+        provider, *_ = self.config.model.name.split(".")
+        return AWSBedrockModelProvider.of_str(provider)
+
+    def _prepare_input(self, payload: dict):
+        prompt, preped = payload.pop("prompt"), {}
+
+        # TODO handle top_p, top_k, etc.
+
+        if self._underlying_provider is AWSBedrockModelProvider.AMAZON:
+            preped["inputText"] = prompt
+            preped["textGenerationConfig"] = rename_payload_keys(
+                payload, {"max_tokens": "maxTokenCount", "stop": "stopSequences"}
+            )
+        elif self._underlying_provider is AWSBedrockModelProvider.ANTHROPIC:
+            if not prompt.startswith("Human: "):
+                prompt = f"Human: {prompt}"
+            if not prompt.endswith("\n\nAssistant:"):
+                prompt = f"{prompt}\n\nAssistant:"
+
+            preped["prompt"] = prompt
+
+            # rename and ensure
+            preped["max_tokens_to_sample"] = max_tokens = payload.pop(
+                "max_tokens", MLFLOW_AI_GATEWAY_ANTHROPIC_DEFAULT_MAX_TOKENS
+            )
+
+            preped["max_tokens_to_sample"] = min(preped["max_tokens_to_sample"], 8191)
+            if max_tokens > MLFLOW_AI_GATEWAY_ANTHROPIC_MAXIMUM_MAX_TOKENS:
+                raise HTTPException(
+                    status_code=422,
+                    detail="Invalid value for max_tokens: cannot exceed "
+                    f"{MLFLOW_AI_GATEWAY_ANTHROPIC_MAXIMUM_MAX_TOKENS}.",
+                )
+
+            preped = {**preped, **rename_payload_keys(payload, {"stop": "stop_sequences"})}
+            if "stop_sequences" not in preped:
+                preped["stop_sequences"] = []
+
+            if "\n\nHuman:" not in preped["stop_sequences"]:
+                preped["stop_sequences"].append("\n\nHuman:")
+        elif self._underlying_provider is AWSBedrockModelProvider.AI21:
+            preped = {
+                "prompt": prompt,
+                **rename_payload_keys(
+                    payload,
+                    {
+                        "stop": "stopSequences",
+                        "candidate_count": "numResults",
+                        "max_tokens": "maxTokens",
+                    },
+                ),
+            }
+
+        elif self._underlying_provider is AWSBedrockModelProvider.COHERE:
+            preped = {
+                "prompt": prompt,
+                **rename_payload_keys(
+                    payload,
+                    {
+                        "stop": "stop_sequences",
+                        "candidate_count": "num_generations",
+                    },
+                ),
+            }
+            # The range of Cohere's temperature is 0-5, but ours is 0-1, so we scale it.
+            if "temperature" in preped:
+                preped["temperature"] *= 5
```

Comment:
Is there an opportunity for us to reuse the payload translation logic from https://github.com/mlflow/mlflow/blob/61d8622dde16578dd844ecc12668580dd81a311c/mlflow/gateway/providers/cohere.py#L43-L44, etc? If possible, could we extract these out into some common utility functions that we can call in each provider (Bedrock for cohere, cohere, bedrock for anthropic, anthropic, etc)?